starfish/pipeline/features/spots/detector/detect.py

```diff
@@ -0,0 +1,211 @@
+
+    Parameters
+    ----------
+    data_image : ImageStack
+        ImageStack containing multiple volumes for which spots' intensities must be calculated
+    spot_attributes : pd.Dataframe
+        Locations and radii of spots
+    measurement_function : Callable[[Sequence], Number]) \
+        Function to apply over the spot volumes to identify the intensity (e.g. max, mean, ...)
+
+    Returns
+    -------
+    IntensityTable :
+        3d tensor of (spot, channel, round) information for each coded spot
+
+    """
+
+    n_ch = data_image.shape[Indices.CH]
+    n_round = data_image.shape[Indices.ROUND]
+    spot_attribute_index = dataframe_to_multiindex(spot_attributes)
+    image_shape: Tuple[int, int, int] = data_image.raw_shape[2:]
+    intensity_table = IntensityTable.empty_intensity_table(
+        spot_attribute_index, n_ch, n_round, image_shape)
+
+    indices = product(range(n_ch), range(n_round))
+    for c, r in indices:
+        image, _ = data_image.get_slice({Indices.CH: c, Indices.ROUND: r})
+        blob_intensities: pd.Series = measure_spot_intensity(
+            image, spot_attributes, measurement_function)
+        intensity_table[:, c, r] = blob_intensities
+
+    return intensity_table
+
+
+def concatenate_spot_attributes_to_intensities(
+        spot_attributes: Sequence[Tuple[SpotAttributes, Dict[Indices, int]]]) \
+        -> IntensityTable:
+    """
+    Merge multiple spot attributes frames into a single IntensityTable without merging across
+    channels and imaging rounds
+
+    Parameters
+    ----------
+    spot_attributes : Sequence[Tuple[SpotAttributes, Dict[Indices, int]]]
+        A sequence of SpotAttribute objects and the Indices (channel, round) that each object is
+        associated with.
+
+    Returns
+    -------
+    IntensityTable :
+        concatenated input SpotAttributes, converted to an IntensityTable object
+
+    """
+    n_ch: int = max(inds[Indices.CH] for _, inds in spot_attributes) + 1
+    n_round: int = max(inds[Indices.ROUND] for _, inds in spot_attributes) + 1
+
+    all_spots = pd.concat([sa.data for sa, inds in spot_attributes])
+    spot_attribute_index = dataframe_to_multiindex(all_spots.drop(['spot_id', 'intensity'], axis=1))
+
+    # TODO ambrosejcarr: remove image_shape
+    z_max = max(max(sa.data['z']) for sa, inds in spot_attributes)
+    y_max = max(max(sa.data['y']) for sa, inds in spot_attributes)
+    x_max = max(max(sa.data['x']) for sa, inds in spot_attributes)
+    image_shape = (z_max, y_max, x_max)
+
+    intensity_table = IntensityTable.empty_intensity_table(
+        spot_attribute_index, n_ch, n_round, image_shape
+    )
+
+    i = 0
+    for attrs, inds in spot_attributes:
+        for _, row in attrs.data.iterrows():
+            intensity_table[i, inds[Indices.CH], inds[Indices.ROUND]] = row['intensity']
+            i += 1
+
+    return intensity_table
+
+
+def detect_spots(
```

Comment:
Implementation of this function, which generalizes spot finding approaches, was the primary goal of this PR.